contributing/pull-request-lifecycle.md

```diff
@@ -0,0 +1,46 @@
+# Pull Request Submission and Lifecycle
```

Comment:
Most of this content is sourced from other repos: [Kubernetes provider](https://raw.githubusercontent.com/hashicorp/terraform-provider-kubernetes/main/_about/PULL_REQUESTS.md), [AWS provider](https://raw.githubusercontent.com/hashicorp/terraform-provider-aws/main/docs/contributing/pullrequest-submission-and-lifecycle.md)